grid_map_core/CMakeLists.txt

```diff
@@ -5,8 +5,7 @@ project(grid_map_core)
 find_package(ament_cmake REQUIRED)
 find_package(grid_map_cmake_helpers REQUIRED)
 
-## Define Eigen addons.
-include(cmake/${PROJECT_NAME}-extras.cmake)
```

Comment:
This needs to be moved down till after the library is created. You can't call `target_compile_definitions` on a target until the target exists.